strats.md

```diff
@@ -454,3 +513,103 @@ A `comeInWithDoorStuckSetup` condition must match with a `leaveWithRunway` condi
+
+A `comeInWithGMode` entrance condition must match with either a `leaveWithGModeSetup` or `leaveWithGMode` entrance condition on the other side of the door:
+- If `mode` is "direct", then it can only match with a `leaveWithGModeSetup`. If `mode` is "indirect", then it can only match with a `leaveWithGMode`.
+
+When matching with a `leaveWithGModeSetup`, a `comeInWithGMode` has implicit requirements:
+- The tech requirement `canEnterGMode`.
+- The `XRayScope` item requirement.
+- A requirement to have at least 1 reserve energy.
+- A requirement to damage down to 0 energy, triggering reserves (causing the reserve energy to become zero and the regular energy to become what the reserve energy was).
+- The requirement `{"or": ["Morph", "canArtificialMorph"]}`, if the `morphed` property of `comeInWithGMode` is true.
+- Requirements for regaining mobility (there may be multiple options here, and they should be treated either as separate strats or the requirements should be joined as though with an `or`):
+  - G-mode mobile: Samus uses the knockback from the previous room to retain mobility. Here the `knockback` property of the `leaveWithGModeSetup` must be true, and reserve energy is assumed to have been drained to at most 4 before doing the strat.
+  - G-mode immobile: Samus uses knockback from a hit in the current room to regain mobility. The possibility of doing this is indicated by the presence of a strat with a `gModeRegainMobility` property with `from` and `to` node matching the `from` node (entrance door node) of the current strat. Any requirements of the `gModeRegainMobility` strat should be included as requirements to execute the current strat; it is important that these requirements be applied *after* the requirement to damage down to 0 energy and trigger reserves. If there are multiple such `gModeRegainMobility` strats, then these should be treated as multiple options.
+
+When matching with a `leaveWithGMode`, a `comeInWithGMode` has an implicit requirement in one scenario:
+- If `morphed` property of `comeInWithGMode` is true but a matching `leaveWithGMode` has `morphed` false, then the `Morph` item is implicitly required.
+
+__Example:__
+```json
+{
+  "name": "G-Mode Morph",
+  "entranceCondition": {
+    "comeInWithGMode": {
+      "mode": "any",
+      "morphed": true
+    }
+  },
+  "requires": []
+}}
+```
+
+
+## G-Mode Regain Mobility
+
+A `gModeRegainMobility` property on a strat indicates that the strat allows Samus to regain mobility (by taking knockback from an enemy or projectile) after entering with G-mode immobile. In all strats with a `gModeRegainMobility` property, the `from` node of the strat must be the same as the `to` node and must be a door node. A strat with `gModeRegainMobility` should normally include an `enemyDamage` requirement in its `requires`.
```

Comment:
also consider removing projectile here

Reply:
>and must be a door node

Pretty minor, but with the Entrance/Exit Condition schema you can GModeImmobile into an entrance node.  That would just be the fall through sand rooms and the sand halls probably.

Reply:
Good point, updated that now here and in a couple other places, to reflect that